### PR TITLE
[bcq-tools] Printing new output arrays for BCQ

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #4337
Full Draft : #4338

This commit will implement new function to pring new output arrays for BCQ

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

**Comment by llFreetimell** [comment]
I wanted to make little bit easier to review, but it was my best :(
If you want, I will explain about this PR at offline... sorry in advance

File: compiler/bcq-tools/generate_bcq_output_arrays [hunk 1]
```diff
@@ -87,7 +87,154 @@ def find_bcq_version(flags):
...
+    # However, operations which do not change original values such as
+    # Ideneity or Transpose can exist between them. In view of TensorFlow Lite,
+    # real users of BCQ applicable constant nodes must be found first.
+    # This work is done by BFS search with queue.
+
+    prefix_node_dict = {}  # key : prefix / value : list of candidates
+    matmul_node_prefix_dict = {}  # key : Name of MatMul node / value : prefix
+
+    queue_prefix = list(prefix_set)
+    queue_nodename = [queue_prefix[idx] + ":0" for idx in range(len(queue_prefix))]
```
Comment (seanshpark) [inline compiler/bcq-tools/generate_bcq_output_arrays:131]: Q) I'm not good at python... what happens if `prefix_set` is empty? is `queue_prefix[idx]` valid?

File: compiler/bcq-tools/generate_bcq_output_arrays [hunk 2]
```diff
@@ -87,7 +87,154 @@ def find_bcq_version(flags):
...
+                    if op.inputs[i].name == nodename:
+                        queue_prefix.append(prefix)
+                        queue_nodename.append(op.outputs[0].name)
+                        break
+
+    # When TensorFlow model is converted to TensorFlow Lite model,
+    # more than one operation can be fused as one.
+    # For example, MatMul + BiasAdd + ReLU in TensorFlow can be fused as
+    # one FullyConnected in TensorFlow Lite.
+    # It means that even real users of BCQ applicable constant nodes
```
Comment (seanshpark) [inline compiler/bcq-tools/generate_bcq_output_arrays:160]: Is `users` here refer to human users? If not, what is the `users` here? maybe `using node(s)` ?

File: compiler/bcq-tools/generate_bcq_output_arrays [hunk 3]
```diff
@@ -87,7 +87,154 @@ def find_bcq_version(flags):
...
+        if op.type == "Const" and "/bcqinfo_" in op.outputs[0].name:
+            # Metadata do not have prefix
+            if "one_compiler/bcqinfo_one_metadata" in op.outputs[0].name:
+                continue
+
+            prefix_index = op.outputs[0].name.index("/bcqinfo_")
+            prefix = op.outputs[0].name[:prefix_index]
+            prefix_set.add(prefix)
+
+            infoname = op.outputs[0].name[prefix_index + 1:-2]
```
Comment (seanshpark) [inline compiler/bcq-tools/generate_bcq_output_arrays:116]: there are several `-2` here and below. can you please add a comment about `-2` ?

File: compiler/bcq-tools/generate_bcq_output_arrays (hunk 1 above)
Comment (llFreetimell) [reply compiler/bcq-tools/generate_bcq_output_arrays:131]: If `prefix_set` is empty, `len(queue_prefix)` becomes 0 and thus for-loop will not be executed :)

File: compiler/bcq-tools/generate_bcq_output_arrays (hunk 2 above)
Comment (llFreetimell) [reply compiler/bcq-tools/generate_bcq_output_arrays:160]: > maybe using node(s)

This is correct, I will fix it :)

File: compiler/bcq-tools/generate_bcq_output_arrays (hunk 3 above)
Comment (llFreetimell) [reply compiler/bcq-tools/generate_bcq_output_arrays:116]: FYI, `op.outputs[0].name` has format like `nodename:0`. `-2` removes `:0`.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**